app/helpers/fhir_helper.rb

```diff
@@ -0,0 +1,329 @@
+  def patient_as_fhir(patient)
+    creator_agent_ref = FHIR::Reference.new
+    creator_app = OauthApplication.where(user_id: patient.creator.id).first
+    if creator_app
+      # Created with an M2M workflow client
+      creator_agent_ref.identifier = FHIR::Identifier.new(value: creator_app.uid)
+      creator_agent_ref.display = creator_app.name
+    else
+      # Created with a user worfklow client
+      creator_agent_ref.identifier = FHIR::Identifier.new(value: patient.creator.id)
+      creator_agent_ref.display = patient.creator.email
+    end
+
+    FHIR::Patient.new(
+      meta: FHIR::Meta.new(lastUpdated: patient.updated_at.strftime('%FT%T%:z')),
+      contained: [FHIR::Provenance.new(
+        # Would like to use a Rails URL Helper here, but we don't get one for this endpoint
+        target: [FHIR::Reference.new(reference: "/fhir/r4/Patient/#{patient.id}")],
+        agent: [
+          FHIR::Provenance::Agent.new(
+            who: creator_agent_ref
+          )
+        ],
+        recorded: patient.created_at.strftime('%FT%T%:z'),
+        activity: {
+          coding: [
+            {
+              system: 'http://terminology.hl7.org/CodeSystem/v3-DataOperation',
+              code: 'CREATE',
+              display: 'create'
+            }
+          ]
+        }
+      )],
+      id: patient.id,
+      active: patient.monitoring,
+      name: [FHIR::HumanName.new(given: [patient.first_name, patient.middle_name].reject(&:blank?), family: patient.last_name)],
+      telecom: [
+        patient.primary_telephone ? FHIR::ContactPoint.new(system: 'phone',
+                                                           value: patient.primary_telephone,
+                                                           rank: 1,
+                                                           extension: [to_string_extension(patient.primary_telephone_type, 'phone-type')])
+                                  : nil,
+        patient.secondary_telephone ? FHIR::ContactPoint.new(system: 'phone',
+                                                             value: patient.secondary_telephone,
+                                                             rank: 2,
+                                                             extension: [to_string_extension(patient.secondary_telephone_type, 'phone-type')])
+                                    : nil,
+        patient.email ? FHIR::ContactPoint.new(system: 'email', value: patient.email, rank: 1) : nil
+      ].reject(&:nil?),
+      birthDate: patient.date_of_birth&.strftime('%F'),
+      address: [
+        FHIR::Address.new(
+          line: [patient.address_line_1, patient.address_line_2].reject(&:blank?),
+          city: patient.address_city,
+          district: patient.address_county,
+          state: patient.address_state,
+          postalCode: patient.address_zip
+        )
+      ],
+      communication: [
+        language_coding(patient.primary_language) ? FHIR::Patient::Communication.new(
```

Comment:
We noticed when getting some data off of prod recently there there are a fair number of primary_language values that are not code-able. Likely from import or old records. Here it looks like we show nothing for those values. Could we have an option where if there's no code for the language it shows the raw string in a field that doesn't have to be coded? Just trying to think through how we avoid missing data when sending this to API users.

Reply:
FHIR really wants this to be coded, but I guess could shoehorn it in there in those cases by just using the `text` property of the `CodeableConcept`. But again, this wasn't actually something I added as part of this task (it's just showing up in this diff because I moved it (context), so I'll create a ticket for this change.

Reply:
https://tracker.codev.mitre.org/browse/SARAALERT-1033